src/backend/columnar/columnar_customscan.c

```diff
@@ -486,14 +489,50 @@ static void
 ColumnarScan_ExplainCustomScan(CustomScanState *node, List *ancestors,
 							   ExplainState *es)
 {
-	TableScanDesc scanDesc = node->ss.ss_currentScanDesc;
+	ColumnarScanDesc scanDesc = (ColumnarScanDesc) node->ss.ss_currentScanDesc;
 
 	if (scanDesc != NULL)
 	{
 		int64 chunkGroupsFiltered = ColumnarScanChunkGroupsFiltered(scanDesc);
 		ExplainPropertyInteger("Columnar Chunk Groups Removed by Filter", NULL,
 							   chunkGroupsFiltered, es);
 	}
+
+	char *projectedColumnsStr = GenerateProjectedColumnStr(node->ss);
+	if (projectedColumnsStr)
+	{
+		ExplainPropertyText("Columnar Projected Columns", projectedColumnsStr, es);
+	}
+}
+
+
+/*
+ * GenerateProjectedColumnStr generates projected column string for explain output.
+ * Returns NULL if no columns are projected.
+ */
+static char *
+GenerateProjectedColumnStr(ScanState scanState)
```

Comment:
Please also take a look at show_sort_group_keys and see if it makes sense to reuse something from there. If not, then this is fine.